### DISCUSSION TITLE
Preloading edge collection's _to and _from

### DISCUSSION THREAD

**Discussion by codewithsisk** [TomGrozev/arangox_ecto#54] [Q&A] [answered]
Is there a way to preload the  _to and _from field when querying the edge collection?  when i query the edge collection i get `#Ecto.Association.NotLoaded<association :from is not loaded>`

**Comment by TomGrozev** [comment]
How are you querying the edge collection? Do you have an example?

You should just be able to use `Repo.preload/3` to preload the fields: https://hexdocs.pm/ecto/Ecto.Repo.html#c:preload/3

**Reply by codewithsisk** [reply]
i have an edge collection called Post, that links _from User _to Topic.  If i want query the Post collection, how do i also get the _from field, i.e. User in my case. I tried `Repo.all(Post)|>Repo.preload(:from)` it return nill on the _from field
`defmodule App.Post do
      use ArangoXEcto.Edge,
      from: App.User,
      to: App.Topic
....
 end
`

**Reply by TomGrozev** [reply]
Sorry @mrsisk  for not getting back to you on this. I have been a combination of busy and travelling.

I have been able to replicate what you are reporting and looking into a fix. Some weird responses in Ecto where it should theoretically work but there seems to be a kink somewhere I haven't been able to find as of yet.

**Reply by TomGrozev** [reply, marked as answer]
This should be resolved in version 2.0. Please check it out.
